lesson16/index.js

```diff
@@ -0,0 +1,120 @@
+
+const createNewsTabs = (allData) => {
+  allData.forEach((content, index) => {
+    const li = document.createElement("li");
+    li.classList.add("tab_item");
+    li.id = content.category;
+    li.textContent = content.category;
+    newsTabs.appendChild(li);
+    content.isActive && li.classList.add("is-active");
+  });
+};
+
+const createNewsContent = (allData) => {
+  // 初期表示用の処理
+  const primaryElement = document.createElement("div");
+  primaryElement.classList.add("news_Wrapper");
+  const lists = document.createElement("ul");
+  lists.classList.add("news_Lists");
+  lists.id = "news_Lists";
+  const primaryImageWrapper = document.createElement("div");
+  primaryElement.appendChild(lists);
+  primaryElement.appendChild(primaryImageWrapper);
+  newsBlock.appendChild(primaryElement);
+
+  const selectedData = allData.find((data) => data.isActive);
+  primaryImageWrapper.classList.add("news_Img");
+  primaryImageWrapper.id = "news_Img";
+  const img = document.createElement("img");
+  img.src = selectedData.image.img;
+  primaryImageWrapper.appendChild(img);
+
+  renderNewsList(selectedData);
+};
+
+const setTabClickEvent = (allData) => {
+  // タブクリック時の処理
+  newsTabs.addEventListener("click", (event) => {
+    // event.targetにis-activeをつける
+    // event.targetのidを取得し、それに紐づく記事と画像を取得
+    const target = event.target;
+
+    // タブのindexを取得し、スタイル変更
+    const activeTab = newsTabs.querySelector(".is-active");
+    activeTab.classList.remove("is-active");
+    target.classList.add("is-active");
+
+    const targetData = allData.find((d) => d.category === target.id);
+    setCategoryImage(targetData);
+    renderNewsList(targetData);
+  });
+};
+
+const setCategoryImage = (targetData) => {
+  const imgWrapper = document.getElementById("news_Img");
+  imgWrapper.innerHTML = "";
+  const img = document.createElement("img");
+  img.src = targetData.image.img;
+  img.alt = targetData.image.alt;
+  imgWrapper.appendChild(img);
+};
+
+const renderNewsList = ({ article }) => {
+  const newsList = document.getElementById("news_Lists");
+  newsList.innerHTML = "";
+  Object.keys(article).forEach((key) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    a.href = article[key].url;
+    a.textContent = article[key].title;
+    const dateDiff = getDateDiff(article[key].publishedDate);
+    if (dateDiff < 4) {
+      const newMark = document.createElement("span");
+      newMark.classList.add("newMark");
+      newMark.innerText = "new!";
+      a.appendChild(newMark);
+    }
+    li.appendChild(a);
+    const commentCount = article[key].comment.length;
+    if (commentCount > 0) {
+      const comment = document.createElement("span");
+      const commentIcon = document.createElement("img");
+      commentIcon.src = "/lesson16/img/icon_comment.png";
+      a.appendChild(commentIcon);
+      comment.textContent = commentCount;
+      li.appendChild(comment);
+    }
```

Comment:
Q
Is there a reason why only span is outside the a tag?
<img width="380" alt="スクリーンショット 2022-02-17 14 29 19" src="https://user-images.githubusercontent.com/43788628/154413024-3b060cc4-2074-442f-b374-00693a2704e1.png">

IMO
And you can cut this out too.
```suggestion
  if (commentCount > 0) {
      a.appendChild(renderComments(commentCount));
    }
```
Cut out function
```js
const renderComments = (commentCount) => {
  const frag = document.createDocumentFragment();
  const comment = document.createElement("span");
  const commentIcon = document.createElement("img");
  commentIcon.src = "/lesson16/img/icon_comment.png";
  frag.appendChild(commentIcon);
  comment.textContent = commentCount;
  frag.appendChild(comment);
  return frag;
};
```

Reply:
[68ae4c9](https://github.com/hippopo-moo/js-handson-work/pull/18/commits/68ae4c9b827be217ad6811f22c4c6a4762ef1fe9)

A.
It didn't make sense to put the comment count span tag outside (I don't remember why I did it myself...).
I think it's better to put it all inside the a-tag, so I fixed it that way!

I also thought it would be better to cut out the process, so I fixed it that way.
The code that Monaka-san provided reminded me of the existence of fragment!